clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -235,6 +229,24 @@ bool CheckedRegionFinder::VisitParmVarDecl(ParmVarDecl *PVD) {
   return true;
 }
 
+bool CheckedRegionFinder::isWild(std::set<ConstraintVariable*> S) { 
+  for (auto Cv : S) 
+    if (Cv->hasWild(Info.getConstraints().getVariables()))
+      return true;
+
+  return false;
+}
+
+bool CheckedRegionFinder::VisitDeclRefExpr(DeclRefExpr* DR) { 
+  auto T = DR->getType();
+  auto D = DR->getDecl();
+
+  if (isWild(Info.getVariable(D, Context)) 
+      || (T->isPointerType() && isUncheckedPtr(T)))
```

Comment:
What's the difference between the above two checks, i.e. `isWild` and `isUncheckedPtr`? Why would the former be false but the latter true, for example?

Reply:
Removed it. This was vestigial due to what was probably an issue with the old constraint system.

Reply:
Seems you still have `isWild(var)` or `(T->isPointerType() && isUncheckedPtr(T))`. It would be useful to know why `isWild` on its own is not sufficient.

Reply:
Renamed it for clarity. Now called `containsCheckedPtr`. It's needed if the type in question is checked but contains unchecked pointers: Ex: `_Ptr<int *>` which isn't allowed in a checked regions.

Reply:
See comment later about function pointer types and what this method might do with them.